src/components/Task.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { RootState } from "../store/store";
+import { Draggable } from "@hello-pangea/dnd";
+import * as styles from "../styles/task.css";
+import Modal from "react-modal"; // Import react-modal
+import { editTask, deleteTask } from "../store/slices/taskSlice"; // Import deleteTask
+import { MdDeleteOutline } from "react-icons/md";
+import { deleteTaskFromList } from "../store/slices/listSlice";
+import { logActivity } from "../store/slices/activityLogSlice";
+
+interface TaskProps {
+  taskId: string;
+  index: number;
+  listId: string;
+}
+
+Modal.setAppElement("#root");
```

Comment:
I feel like you are misusing this. I haven't checked the docs but I assume you either have to set the root once or set it per component with the component itself being the root.

Reply:
Judging by the method name, I assume it's the former.